File: opentsdb-proxy/src/main/java/net/opentsdb/proxy/clients/ProxyClient.java
package net.opentsdb.proxy.clients;

import com.google.inject.Guice;
import com.google.inject.Inject;
import com.google.inject.Injector;
import net.opentsdb.client.netty.TsdbClient;
import net.opentsdb.client.netty.modules.TsdbClientModule;

import java.util.Properties;

public class ProxyClient implements Client {
  private final TsdbClient tsdbClient;

  @Inject
  public ProxyClient(Properties properties) {
    Injector injector = Guice.createInjector(new TsdbClientModule(properties));

    this.tsdbClient = injector.getInstance(TsdbClient.class);
    this.tsdbClient.startAndWait();
  }

  @Override
  public void sendMessage(String message) {
    tsdbClient.send(message);
  }
}

File: opentsdb-proxy/src/main/scala/net/opentsdb/proxy/clients/ProxyClient.scala
package net.opentsdb.proxy.clients

import com.google.inject.{Guice, Inject}
import java.util.Properties
import net.opentsdb.client.netty.modules.TsdbClientModule
import net.opentsdb.client.netty.TsdbClient

class ProxyClient @Inject() (props: Properties) extends Client {
  val injector = Guice.createInjector(new TsdbClientModule(props))

  val tsdbclient = injector.getInstance(classOf[TsdbClient])
  tsdbclient.startAndWait()

  def sendMessage(message: String) {
    tsdbclient.send(message)
  }
}
